tutorials/ner_pytorch_medical/scripts/torch_ner_model.py

```diff
@@ -0,0 +1,143 @@
+from typing import Optional, List
+from thinc.api import (
+    with_array,
+    chain,
+    Model,
+    PyTorchWrapper,
+    PyTorchLSTM,
+)
+from thinc.types import Floats2d
+
+from spacy.tokens import Doc
+from spacy.util import registry
+import torch
+from torch import nn
+
+
+@registry.architectures("TorchBiLSTMEncoder.v1")
+def BiLSTMEncoder(
+    width: int, depth: int, dropout: float
+) -> Model[List[Floats2d], List[Floats2d]]:
+    """Encode context using bidirectonal LSTM layers. Requires PyTorch.
+    width (int): The input and output width. These are required to be the same,
+        to allow residual connections. This value will be determined by the
+        width of the inputs. Recommended values are between 64 and 300.
+    depth (int): The number of recurrent layers.
+    dropout (float): Creates a Dropout layer on the outputs of each LSTM layer
+        except the last layer. Set to 0 to disable this functionality.
+    """
+    return PyTorchLSTM(width, width, bi=True, depth=depth, dropout=dropout)
+
+
+@registry.architectures("TorchEntityRecognizer.v1")
+def build_torch_ner_model(
+    tok2vec: Model[List[Doc], List[Floats2d]],
+    hidden_width: int,
+    dropout: Optional[float] = None,
+    nO: Optional[int] = None,
+) -> Model[List[Doc], List[Floats2d]]:
+    """Build a tagger model, using a provided token-to-vector component. The tagger
+    model simply adds a linear layer with softmax activation to predict scores
+    given the token vectors.
+    tok2vec (Model[List[Doc], List[Floats2d]]): The token-to-vector subnetwork.
+    nO (int or None): The number of tags to output. Inferred from the data if None.
+    RETURNS (Model[List[Doc], List[Floats2d]]): Initialized Model
+    """
+    t2v_width = tok2vec.get_dim("nO") if tok2vec.has_dim("nO") else 768
```

Comment:
This 768 here is kind of dodgy. Is it necessary? Also the `maybe_get_dim` method is better here.
```suggestion
    t2v_width = tok2vec.maybe_get_dim("nO")
```

Reply:
Definitely super dodgy :). This was there for my Transformer testing, just forgot to actually fix it. Seems the dim nO isn't set on the tok2vec but on the listener and it only gets set after init based on the hidden_size of the huggingface config for the model. So it's a slightly involved change to make this work dynamically but this should work now.